src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -866,6 +909,31 @@ public override bool TryGetArgumentByNameOrIndex(string? name, int position, Typ
 
                 return false;
             }
+
+            public override bool TryGetTopLevelProperty<T>(string name, [MaybeNull] out T value)
+            {
+                value = default;
+
+                JToken? serializedValue = null;
+                if (this.topLevelProperties?.TryGetValue(name, out serializedValue) is true)
```

Comment:
Aren't we going to add this capability to all message types? I thought we talked about adding version information from the RPC server to error messages, for example.

Reply:
New changes add this to all message types, unfortunately the fact that we can't create a base class for a custom message type caused some code repetition but I've tried to have no new object allocations as much as possible in the code paths where these methods are not used.